lighthouse-core/lib/network-recorder.js

```diff
@@ -168,15 +165,17 @@ class NetworkRecorder extends EventEmitter {
   }
 
   /**
-   * Construct network records from a log of devtools protocol messages.
-   * @param {!DevtoolsLog} log
+   * Construct network records from a log of devtools protocol messages. If a
+   * driver is provided, it will be used for request content lookups.
+   * @param {!DevtoolsLog} devtoolsLog
+   * @param {!Driver=} driver
```

Comment:
> unrelated, but I still vote for a driver helper function instead, this is a really annoying artifact of having it be the method on network records :(

the problem is we need a live `driver` to get resource content, which will only be accessible in gatherers, but we need to create records from a devtools log during audits too (via a computed artifact) when there is no live driver. The first one argues for an instance method on `driver`, the second argues for a static method. So...it could be a static `driver` method, but still take a `driver` instance? But that seems like it has the same issue.

Not sure if there's a good solution :S